File: SDKs/art19.com/1.0.0/python/src/sdk/models/operations/get_images_id_.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import image as shared_image
from ..shared import resourcelink as shared_resourcelink


@dataclasses.dataclass
class GetImagesIDPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass_json
@dataclasses.dataclass
class GetImagesID200ApplicationVndAPIPlusJSON:
    data: Optional[shared_image.Image] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    links: Optional[shared_resourcelink.ResourceLink] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('links') }})
    

@dataclasses.dataclass
class GetImagesIDRequest:
    path_params: GetImagesIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetImagesIDResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    get_images_id_200_application_vnd_api_plus_json_object: Optional[GetImagesID200ApplicationVndAPIPlusJSON] = dataclasses.field(default=None)
    

File: SDKs/art19.com/1.0.0/typescript/src/sdk/models/operations/getimagesid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetImagesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetImagesId200ApplicationVndApiPlusJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data?: shared.Image;

  @SpeakeasyMetadata({ data: "json, name=links" })
  links?: shared.ResourceLink;
}


export class GetImagesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetImagesIdPathParams;
}


export class GetImagesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getImagesId200ApplicationVndApiPlusJsonObject?: GetImagesId200ApplicationVndApiPlusJson;
}
